<a href="https://colab.research.google.com/github/sulaimanbehzad/Clustering_according_to_item_popularity/blob/main/clustring.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Clustering of Digikala dataset based on items popularity  


In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.preprocessing import LabelEncoder, normalize

### Cloning the github repository in colab

In [ ]:
% cd /content/drive/MyDrive/Clustering_according_to_item_popularity/


/content/drive/MyDrive/Clustering_according_to_item_popularity


In [ ]:
! ls

clustring.ipynb  LICENSE  README.md


In [ ]:
! git clone https://github.com/sulaimanbehzad/Clustering_according_to_item_popularity.git

Cloning into 'Clustering_according_to_item_popularity'...
remote: Enumerating objects: 8, done.
remote: Counting objects: 100% (8/8), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 8 (delta 1), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (8/8), done.


In [ ]:
! git status

On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean


In [ ]:
! git pull

remote: Enumerating objects: 5, done.
remote: Counting objects: 100% (5/5), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 3 (delta 1), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (3/3), done.
From https://github.com/sulaimanbehzad/Clustering_according_to_item_popularity
   3c88399..1f2eeea  main       -> origin/main
Updating 3c88399..1f2eeea
Fast-forward
 clustring.ipynb | 170 +++++++++++++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 168 insertions(+), 2 deletions(-)


In [ ]:
! git push

### Reading and preprocessing the dataset

In [4]:
df = pd.read_excel('/content/drive/MyDrive/Clustering_according_to_item_popularity/keifiat.xlsx')

In [5]:
df.head()

product_id  ...                                      disadvantages
0        3692  ...                                                NaN
1       90213  ...                                          ["ندارد"]
2       59473  ...                                                NaN
3      120499  ...  ["کیفیت پایین اسپیکر\r","حاشیه خالی زیر صفحه ن...
4       67200  ...                                    ["ظرفیت متوسط"]

[5 rows x 12 columns]

In [6]:
df.describe()

product_id       user_id          likes       dislikes
count  100000.000000  1.000000e+05  100000.000000  100000.000000
mean   426482.510140  4.642959e+06       3.114790       1.649460
std    232470.511159  2.619034e+06       8.050031       8.912705
min       300.000000  4.661320e+05       0.000000       0.000000
25%    220871.250000  2.613059e+06       0.000000       0.000000
50%    427858.000000  4.710290e+06       1.000000       0.000000
75%    619397.250000  7.094806e+06       4.000000       1.000000
max    909423.000000  8.623288e+06     854.000000    1344.000000

Select the attributes that will give information about the popularity of the item

In [9]:
df1 = df[['product_id', 'likes', 'dislikes','comment', 'recommend']]

Check the dataframe for null values and handle them

In [10]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column      Non-Null Count   Dtype 
---  ------      --------------   ----- 
 0   product_id  100000 non-null  int64 
 1   likes       100000 non-null  int64 
 2   dislikes    100000 non-null  int64 
 3   comment     99883 non-null   object
 4   recommend   100000 non-null  object
dtypes: int64(3), object(2)
memory usage: 3.8+ MB


In [15]:
df1.isnull().sum()

product_id      0
likes           0
dislikes        0
comment       117
recommend       0
dtype: int64

looking at the database 'recommend' we can see that \N values also represent null values which wouldn't be considered in the above

In [18]:
df1['recommend'].value_counts()

recommended        36972
\N                 36382
not_recommended    16110
no_idea            10536
Name: recommend, dtype: int64

we can see that beside the 117 missing values we have 36382 records with \N values
